### PR TITLE
Adding samples and few options to the runner script

### PR DESCRIPTION
**Pull request by cippy**
Adding Top and Diboson samples from cmgtools, to compare histograms with it.
Also adding options to remove muon corrections and/pr Scetlib ones, since they were not in cmgtools.

The part about samples also uses an update to be pushed to narf, adding a group element in the class Dataset in **narf/dataset.py**
These changes to narf are not yet pushed but are in the attached image. The are very simple and probably it might be easier to include them directly in https://github.com/bendavid/narf.git and then merge from it.
The idea behind this update is to allow for minimal interaction with **wremnants/datasets/datagroups.py** when updating **wremnants/datasets/datasets2016.py** with a new sample, so that it can be automatically included in an existing group already defined in datagroups.py, otherwise one might fail to edit both scripts consistently. The feature is not mandatory, though.
The other change in **narf/graph_builder.py** is just for better debugging, to see what those printed numbers are associated to.

image

### CODE REVIEW

**Comment by kdlong** [comment]
Sorry for being so slow on this. Did the data group addition make it into narf?

File: scripts/histmakers/mw_with_mu_eta_pt.py [hunk 1]
```diff
@@ -2,8 +2,12 @@
 
 parser = argparse.ArgumentParser()
 parser.add_argument("-j", "--nThreads", type=int, help="number of threads", default=None)
+parser.add_argument("-e", "--era", type=str, choices=["2016PreVFP","2016PostVFP"], help="Data set to process", default="2016PostVFP")
```
Comment (kdlong) [inline scripts/histmakers/mw_with_mu_eta_pt.py:5]: Probably should have a full 2016 option?

File: wremnants/datasets/datagroups.py [hunk 2]
```diff
@@ -155,13 +155,15 @@ def __init__(self, infile, combine=False, wlike=False):
                     signalOp = sel.signalHistWmass,
                 ),
                 "Top" : dict(
-                    members = [self.datasets["TTSemileptonicPostVFP"], self.datasets["TTLeptonicPostVFP"]],
+                    #members = [self.datasets["TTSemileptonicPostVFP"], self.datasets["TTLeptonicPostVFP"]],
```
Comment (kdlong) [inline wremnants/datasets/datagroups.py:158]: Just delete commented out lines (also below)

File: scripts/histmakers/mw_with_mu_eta_pt.py [hunk 3]
```diff
@@ -138,14 +150,15 @@ def build_graph(df, dataset):
         df = df.Define("weight_fullMuonSF_withTrackingReco", muon_efficiency_helper, ["goodMuons_pt0", "goodMuons_eta0", "goodMuons_charge0", "passIso"])
         df = df.Define("weight_newMuonPrefiringSF", muon_prefiring_helper, ["Muon_correctedEta", "Muon_correctedPt", "Muon_correctedPhi", "Muon_looseId"])
 
-        applyScetlibCorr = True
```
Comment (kdlong) [inline scripts/histmakers/mw_with_mu_eta_pt.py:141]: I don't understand if you actually use the noScetlibCorr argument. Shouldn't it be used here? There is also a conflict that I introduced (sorry) related to these lines of code

File: scripts/histmakers/mw_with_mu_eta_pt.py (hunk 1 above)
Comment (cippy) [reply scripts/histmakers/mw_with_mu_eta_pt.py:7]: Yeah, better to wait, adding that would imply other changes which for now would just make the code messier.

File: wremnants/datasets/datagroups.py (hunk 2 above)
Comment (cippy) [reply wremnants/datasets/datagroups.py]: Done now

File: scripts/histmakers/mw_with_mu_eta_pt.py (hunk 3 above)
Comment (cippy) [reply scripts/histmakers/mw_with_mu_eta_pt.py]: It was changed to a command line option, anyway the conflict should be fixed now

**Comment by kdlong** [comment]
Sorry for being slow on this, do you mind updating narf and fixing the merge conflict I created?

**Comment by cippy** [comment]
narf has been already updated since last week, sorry if I didn't make it clear. 
I made a PR to Josh's repository. When you merge from that you should get everything

**Comment by cippy** [comment]
Should be ok now

**Pull request merged**